src/block/serverside.php

```diff
@@ -0,0 +1,191 @@
+ *
+ * @param WP_REST_Request $request
+ *
+ * @return WP_Error|WP_REST_Response
+ */
+function yelp_api_rest_callback( WP_REST_Request $request ) {
+
+    // Get parameters from request
+    $params = $request->get_params();
+
+    if ( array_key_exists( 'term', $params ) && array_key_exists( 'location', $params ) || isset( $params['keyValidation'] ) ) {
+        return yelp_retrieve_business_search_results( $params );
+    } else {
+        return yelp_retrieve_business_details( $params );
+    }
+
+
+}
+
+/**
+ * @param $params
+ *
+ * @return WP_Error|WP_REST_Response
+ */
+function yelp_retrieve_business_search_results( $params ) {
+    $apiKey = get_option( 'yelp_widget_settings' );
+    $apiKey = $params['apiKey'] ?? $apiKey['yelp_widget_fusion_api'];
+
+    if ( empty( $params['apiKey'] ) && isset( $params['keyValidation'] ) ) {
+        return new WP_Error( 'yelp_api_error', esc_html__( 'Request must include a valid Yelp Fusion API key.', 'yelp-widget-pro' ), [ 'status' => 400 ] );
+    }
+
+    // Ready to Query the Yelp API.
+    $args = [
+        'headers' =>
+            [
+                'user-agent'    => '',
+                'authorization' => 'Bearer ' . $apiKey,
+            ],
+    ];
+
+    // Get business search data.
+    $requestUrl = add_query_arg( [
+        'term'     => $params['term'] ?? '',
+        'location' => $params['location'] ?? 'NYC',
+    ], 'https://api.yelp.com/v3/businesses/search' );
+
+    $requestRequest = wp_remote_get( $requestUrl, $args );
+    $requestBody    = json_decode( wp_remote_retrieve_body( $requestRequest ) );
+
+    if ( $requestBody->error ) {
+        return new WP_Error( 'yelp_api_error', $requestBody->error->description, [ 'status' => 400 ] );
+    }
+
+    // Create the response object
+    return new WP_REST_Response( wp_json_encode( $requestBody->businesses ), 200 );
+}
+
+/**
+ * @param $params
+ *
+ * @return string|WP_REST_Response
+ */
+function yelp_retrieve_business_details( $params ) {
+
+    // Check if transient exists.
+    $business_details = get_transient( $params['businessId'] );
+    if ( $business_details ) {
+        return $business_details;
+    } else {
+
+        $apiKey = get_option( 'yelp_widget_settings' );
+        $apiKey = $apiKey['yelp_widget_fusion_api'];
+
+        // Ready to Query the Yelp API.
+        $args = [
+            'headers' =>
+                [
+                    'user-agent'    => '',
+                    'authorization' => 'Bearer ' . $apiKey,
+                ],
+        ];
+
+        $businessDetailsReq = add_query_arg( [
+            'locale' => $params['locale'] ?? 'en_US',
+        ], "https://api.yelp.com/v3/businesses/{$params['businessId']}" );
+
+        // 1️⃣ Get business details.
+        $response        = wp_safe_remote_get( $businessDetailsReq, $args );
+        $businessDetails = json_decode( wp_remote_retrieve_body( $response ) );
+
+        // 2⃣ Get business reviews.
```

Comment:
Typo? I think you mean this to be an emoji

Reply:
Fixed